src/envoy/http/mixer/report_data.h

```diff
@@ -48,12 +48,12 @@ class ReportData : public ::istio::control::http::ReportData {
       ::istio::control::http::ReportData::ReportInfo* data) const override {
     data->received_bytes = info_.bytesReceived();
     data->send_bytes = info_.bytesSent();
-    data->duration =
-        std::chrono::duration_cast<std::chrono::nanoseconds>(info_.duration());
+    data->duration = std::chrono::duration_cast<std::chrono::nanoseconds>(
+        std::chrono::system_clock::now() - info_.startTime());
```

Comment:
Could you please change it back to info_.duration()? 
duration() returns the microseconds duration of the first byte received to the last byte sent.
And this duration is used to generate "response.duration" attribute, which is defined as "The amount of time the response took to generate."

Reply:
duration() is no longer an API. Only startTime() is available.

Reply:
Use startTimeMonotonic and `chrono::steady_clock` to calculate the duration.